### PR TITLE
feat(discover2): Improve UX for naming Saved Queries

### PR DESCRIPTION
**Pull request by leedongwei**
This PR combines `saveQueryButton` with `savedQueryButtonGroup`. The main challenge was to maintain the state using `Location` and allowing the state to be changed by 2 different `<Input/>` without any delay on each keystroke.

@dashed @davidenwang I've checked that the parts for analytics are correct but you'll want to double-check on them.

Refs SEN-1167

### CODE REVIEW

**Comment by dashed** [comment]
@leedongwei 

When playing with it, I think I found a bug. I haven't dug too deep into the pull-request to see the cause. 

Opening a pre-built query, and attempting to rename it and dismiss it will cause the name to be blank

Kapture 2019-11-04 at 13 18 22

File: tests/js/spec/views/eventsV2/savedQueryButtonGroup.spec.jsx [hunk 1]
```diff
@@ -2,11 +2,12 @@ import React from 'react';
...
 import {browserHistory} from 'react-router';
 
-import EventSaveQueryButton from 'app/views/eventsV2/saveQueryButton';
+import SavedQueryButtonGroup from 'app/views/eventsV2/savedQueryButtonGroup';
 import {ALL_VIEWS} from 'app/views/eventsV2/data';
 import EventView from 'app/views/eventsV2/eventView';
 
-describe('EventsV2 > SaveQueryButton', function() {
+// eslint-disable-next-line jest/no-disabled-tests
+describe.skip('EventsV2 > SaveQueryButtonGroup', function() {
```
Comment (dashed) [inline tests/js/spec/views/eventsV2/savedQueryButtonGroup.spec.jsx]: @leedongwei hopefully we can restore and polish these tests up soon in a follow up PR 👍

**Comment by leedongwei** [comment]
@dashed Good catch on the disappearing values. I've fixed it in the latest commits.

File: tests/js/spec/views/eventsV2/savedQueryButtonGroup.spec.jsx (hunk 1 above)
Comment (leedongwei) [reply tests/js/spec/views/eventsV2/savedQueryButtonGroup.spec.jsx]: Yep, I'm following up with tests in a separate PR but that's probably coming later in the afternoon.

File: src/sentry/static/sentry/app/views/eventsV2/index.tsx [hunk 2]
```diff
@@ -80,29 +98,16 @@ class EventsV2 extends React.Component<Props> {
...
-
-    return [t('Discover')];
-  };
-
   render() {
     const {organization, location, router} = this.props;
+
     const eventSlug = getFirstQueryString(location.query, 'eventSlug');
     const eventView = EventView.fromLocation(location);
+    const queryName = getFirstQueryString(location.query, 'name', t('Untitled query'));
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/index.tsx]: @leedongwei The `name` query string should be already derived from `EventView.fromLocation(location)`.

The name should be accessible from `eventView.name`.

Comment (leedongwei) [reply src/sentry/static/sentry/app/views/eventsV2/index.tsx]: Resolved with the latest force-push.

File: src/sentry/static/sentry/app/views/eventsV2/index.tsx [hunk 3]
```diff
@@ -112,8 +107,21 @@ class EventsV2 extends React.Component<Props> {
               <NoProjectMessage organization={organization}>
                 <PageHeader>
                   <PageHeading>
-                    {pageTitle} <BetaTag />
+                    Discover
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/index.tsx]: @leedongwei `{t('Discover')}` here

File: src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  location: Location;
+  organization: Organization;
+  eventView: EventView;
+  savedQueries: SavedQuery[];
+};
+type State = {
+  isNewQuery: boolean;
+  isEditingQuery: boolean;
+
+  isTyping: boolean;
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx]: @leedongwei it looks like this state prop `isTyping` is unused

**Review by dashed — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx [hunk 5]
```diff
@@ -53,8 +53,14 @@ class SavedQueryButtonGroup extends React.Component<Props> {
...
 
+    if (savedQueriesLoading) {
+      return prevState;
+    }
+
+    const savedEventView = EventView.fromSavedQuery(savedQuery);
+
     // Switching from a SavedQuery to another SavedQuery
-    if (queryId !== nextEventView.id) {
+    if (savedEventView.id !== nextEventView.id) {
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx]: @leedongwei this one is slightly subtle. I've opted to use `savedEventView.id` to make it more explicit.

**Review by dashed — APPROVED** [review]
Looks good to me 👍 

Just need to rebase and it's good to go.

**Comment by leedongwei** [comment]
Rebase + fix merge conflicts + autosquash.

I gave it a quick read through and it seems alright, I'll merge tomorrow morning.

**Comment by dashed** [comment]
The rebase looks good to me 👍

**Pull request merged**